internal/proxy/proxy.go

```diff
@@ -282,6 +287,29 @@ func NewClient(ctx context.Context, cmd *cobra.Command, conf *Config) (*Client,
 	return &Client{mnts: mnts, cmd: cmd, dialer: d}, nil
 }
 
+// dialOptions interprets appropriate dial options for a particular instance
+// configuration
+func dialOptions(conf *Config, inst *InstanceConnConfig) []cloudsqlconn.DialOption {
+	var opts []cloudsqlconn.DialOption
+
+	// related to --auto-iam-authn=(bool)
+	if inst.IAMAuthN != nil {
+		opts = append(opts, cloudsqlconn.WithDialIAMAuthN(*inst.IAMAuthN))
+	} else {
+		opts = append(opts, cloudsqlconn.WithDialIAMAuthN(conf.IAMAuthN))
+	}
+
+	// related to --private-ip=(bool) option
+	if inst.PrivateIP != nil && *inst.PrivateIP ||
+		inst.PrivateIP == nil && conf.PrivateIP {
+		opts = append(opts, cloudsqlconn.WithPrivateIP())
+	} else {
+		opts = append(opts, cloudsqlconn.WithPublicIP())
```

Comment:
We default to public, so this else block can probably be deleted.

Reply:
Even though WithPublicIP() is the default for the cloudsqlconn library today, I would rather make it explicit in our code. Too often, the upstream library team changes a default and unexpectedly breaks something. Are you ok if I leave this as-is?